core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/crt/internal/signer/AwsChunkedV4aPayloadSigner.java

```diff
@@ -103,12 +106,7 @@ public ContentStreamProvider sign(ContentStreamProvider payload, V4aContext v4aC
      * signature is dependent on the request signature ("seed" signature).
      */
     private void setupSigExt(ChunkedEncodedInputStream.Builder builder, RollingSigner rollingSigner) {
-        builder.addExtension(
-            chunk -> Pair.of(
-                "chunk-signature".getBytes(StandardCharsets.UTF_8),
-                rollingSigner.sign(chunk)
-            )
-        );
+        builder.addExtension(new SigV4aChunkExtensionProvider(rollingSigner, credentialScope));
```

Comment:
I like the breakout of the chunk logic to their respective providers. Now this method and other one liners don't seem to provide much in terms of readability, can we just fold back the code to the `sign` method?